xarray/core/dataset.py

```diff
@@ -2648,6 +2648,119 @@ def stack(self, dimensions=None, **dimensions_kwargs):
             result = result._stack_once(dims, new_dim)
         return result
 
+    def to_stacked_array(self, new_dim, sample_dims, variable_dim='variable',
+                         name=None):
+        """Combine variables of differing dimensionality into a DataArray
+        without broadcasting.
+
+        This function is basically version of Dataset.to_array which does not
+        broadcast the variables.
```

Comment:
I think there is a typo: 'a version'

maybe reformulate to
 This function is similar to Dataset.to_array but does not broadcast the variables.

Reply:
Okay. Changed to "This _method_ is similar to Dataset.to_array but does not broadcast variables."